src/lib/routes/admin-api/api-token.ts

```diff
@@ -16,48 +16,132 @@ import User from '../../types/user';
+} from '../../openapi/spec/api-token-schema';
+import { emptyResponse } from '../../openapi/spec/empty-response';
 
-interface IServices {
-    apiTokenService: ApiTokenService;
-    accessService: AccessService;
-}
-
-class ApiTokenController extends Controller {
+export class ApiTokenController extends Controller {
     private apiTokenService: ApiTokenService;
 
     private accessService: AccessService;
 
+    private openApiService: OpenApiService;
+
     private logger: Logger;
 
-    constructor(config: IUnleashConfig, services: IServices) {
+    constructor(
+        config: IUnleashConfig,
+        {
+            apiTokenService,
+            accessService,
+            openApiService,
+        }: Pick<
+            IUnleashServices,
+            'apiTokenService' | 'accessService' | 'openApiService'
+        >,
+    ) {
         super(config);
-        this.apiTokenService = services.apiTokenService;
-        this.accessService = services.accessService;
+        this.apiTokenService = apiTokenService;
+        this.accessService = accessService;
+        this.openApiService = openApiService;
         this.logger = config.getLogger('api-token-controller.js');
 
-        this.get('/', this.getAllApiTokens, READ_API_TOKEN);
-        this.post('/', this.createApiToken, CREATE_API_TOKEN);
-        this.put('/:token', this.updateApiToken, UPDATE_API_TOKEN);
-        this.delete('/:token', this.deleteApiToken, DELETE_API_TOKEN);
+        this.route({
+            method: 'get',
+            path: '',
+            handler: this.getAllApiTokens,
+            permission: READ_API_TOKEN,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'getAllApiTokens',
+                    responses: {
+                        200: createResponseSchema('apiTokensSchema'),
+                    },
+                }),
+            ],
+        });
+
+        this.route({
+            method: 'post',
+            path: '',
+            handler: this.createApiToken,
+            permission: CREATE_API_TOKEN,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'createApiToken',
+                    requestBody: createRequestSchema('createApiTokenSchema'),
+                    responses: {
+                        201: createResponseSchema('apiTokenSchema'),
+                    },
+                }),
+            ],
+        });
+
+        this.route({
+            method: 'put',
+            path: '/:token',
+            handler: this.updateApiToken,
+            permission: UPDATE_API_TOKEN,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'updateApiToken',
+                    responses: {
+                        200: emptyResponse,
+                        400: emptyResponse,
```

Comment:
Could leave out `400` for now. We should probably find a generic way to spec 400 errors.